--- a/functions	2019-06-30 23:34:08.415624900 +0900
+++ b/functions	2019-06-30 23:36:30.831114400 +0900
@@ -514,7 +514,7 @@ backup_dump_directory() {
   local extension=${2}
 
   echo "Dumping ${dirname}..."
-  exec_as_redmine tar -caf ${config_default_backup_storage_path}/${dirname}${extension} -C ${directory} .
+  exec_as_redmine tar -caf ${config_default_backup_storage_path}/${dirname}${extension} -C ${directory} . || [[ $? == 1 ]]
 }
 
 backup_dump_information() {
@@ -534,7 +534,7 @@ backup_create_archive() {
   eval $(parse_yaml ${REDMINE_CONFIG} config_)
 
   echo "Creating backup archive: ${tar_file}..."
-  exec_as_redmine tar -caf ${config_default_backup_storage_path}/${tar_file} -C ${config_default_backup_storage_path} $@
+  exec_as_redmine tar -caf ${config_default_backup_storage_path}/${tar_file} -C ${config_default_backup_storage_path} $@ || [[ $? == 1 ]]
   exec_as_redmine chmod 0644 ${config_default_backup_storage_path}/${tar_file}
 
   for f in $@
